lib/retriable/config.rb

```diff
@@ -1,29 +1,43 @@
+require_relative "exponential_backoff"
+
 module Retriable
   class Config
-    attr_accessor :sleep_disabled
-    attr_accessor :tries
-    attr_accessor :base_interval
-    attr_accessor :max_interval
-    attr_accessor :rand_factor
-    attr_accessor :multiplier
-    attr_accessor :max_elapsed_time
-    attr_accessor :intervals
-    attr_accessor :timeout
-    attr_accessor :on
-    attr_accessor :on_retry
+    ATTRIBUTES = ExponentialBackoff::ATTRIBUTES + [
+      :sleep_disabled,
+      :max_elapsed_time,
+      :intervals,
+      :timeout,
+      :on,
+      :on_retry,
+    ].freeze
+
+    attr_accessor(*ATTRIBUTES)
+
+    def initialize(opts = {})
+      backoff = ExponentialBackoff.new
+
+      @tries            = backoff.tries
+      @base_interval    = backoff.base_interval
+      @max_interval     = backoff.max_interval
+      @rand_factor      = backoff.rand_factor
+      @multiplier       = backoff.multiplier
+      @sleep_disabled   = false
+      @max_elapsed_time = 900 # 15 min
+      @intervals        = nil
+      @timeout          = nil
+      @on               = [StandardError]
+      @on_retry         = nil
+
+      opts.each do |k, v|
+        raise ArgumentError, "#{k} is not a valid option" if !ATTRIBUTES.include?(k)
```

Comment:
I've never really preferred `unless`. I've personally just never found it more readable, for me at least, than `if !`.